uber/site_sections/summary.py

```diff
@@ -175,12 +175,115 @@ def food_eligible(self, session):
+                    model_instance = getattr(session, selected_model)(id, allow_invalid=True)
+                except:
+                    session.rollback()
+                    # otherwise, make a new one and add it to the session for when we commit
+                    model_instance = model()
+                    session.add(model_instance)
+
+            for colname, val in row.items():
+                col = cols[colname]
+                if not val:
+                    # in a lot of cases we'll just have the empty string, so we'll just
+                    # do nothing for those cases
+                    continue
+                if isinstance(col.type, Choice):
+                    # the export has labels, and we want to convert those back into their
+                    # integer values, so let's look that up (note: we could theoretically
+                    # modify the Choice class to do this automatically in the future)
+                    label_lookup = {val: key for key, val in col.type.choices.items()}
+                    val = label_lookup[val]
+                elif isinstance(col.type, MultiChoice):
+                    # the export has labels separated by ' / ' and we want to convert that
+                    # back into a comma-separate list of integers
+                    label_lookup = {val: key for key, val in col.type.choices}
+                    vals = [label_lookup[label] for label in val.split(' / ')]
+                    val = ','.join(map(str, vals))
+                elif isinstance(col.type, UTCDateTime):
+                    # we'll need to make sure we use whatever format string we used to
+                    # export this date in the first place
+                    try:
+                        val = UTC.localize(datetime.strptime(val, date_format + ' %H:%M:%S'))
+                    except:
+                        val = UTC.localize(datetime.strptime(val, date_format))
+                elif isinstance(col.type, Date):
+                    val = datetime.strptime(val, date_format).date()
+                elif isinstance(col.type, Integer):
+                    val = int(val)
+
+                # now that we've converted val to whatever it actually needs to be, we
+                # can just set it on the model
+                setattr(model_instance, colname, val)
+
+            try:
+                session.commit()
+            except:
+                log.error('ImportError', exc_info=True)
+                session.rollback()
+                message = 'Import unsuccessful'
+
+            id_list.append(model_instance.id)
+
+        all_instances = session.query(model).filter(model.id.in_(id_list)).all() if id_list else None
+
+        if model == Attendee:
+            form_link = '../registration/form'
+        elif model == Group:
+            form_link = '../groups/form'
+        elif model == Job:
+            form_link = '../jobs/form'
+        else:
+            form_link = None
```

Comment:
Anytime we see this kind of if/elif chain, we should probably replace it with some dictionary or class hierarchy somewhere.  Two possible suggestions:

1) We make a new field on the ``MagModel`` class.  Something like a ``form_link`` field on the ``MagModel`` base class which is ``None`` and then the ``Attendee`` class sets it to ``../registration/form`` and the Job class sets it to ``../jobs/form`` etc.  Then we just say ``form_link = model.form_link`` or whatever.

2) Option one involves mixing our data layer with our presentation layer, so maybe we avoid that and instead have a dictionary somewhere like

```python
form_links = {
    Attendee: '../registration/form',
    ...
}
```
then we can just say ``form_link = form_links.get(model)``.

Either way this isn't a huge deal since this is currently only used in one place.  Though with that being said, there's already a ``|form_link`` filter which currently only works for attendees, and it would be nice to make it more generic, since there are a LOT of places where we kind of hard-code form links.

Feel free to ignore this suggestion for now, since requiring that kind of refactor as part of this feature might be a bit out of scope for this PR.  On the other hand, if you feel like biting the bullet now and taking this opportunity to put a better approach for form links into place, that'd be pretty sweet.